docs/tutorials/send-payjoin-with-pdk.md

```diff
@@ -0,0 +1,311 @@
+    As you would guess, the `https_agent` function is used in order to make
+    https requests, the sender will use this function to send the payjoin
+    request to the reciever.
+
+    Here, we are using `danger_accept_invalid_certs` as we running this locally,
+    in production this should be initiated with valid certificates only.
+
+```rust
+ fn https_agent() -> reqwest::Client {
+     let https = reqwest::Client::builder()
+         .danger_accept_invalid_certs(true)
+         .build()
+         .expect("failed to build https client");
+     https
+ }
+```
+
+  * Add `Sender` struct
+    
+    We will add an implementation to the struct with a single function,
+    `try_payjoin`. The function will take a `payjoin::Uri` as an input and
+    attempt to perform a payjoin transaction.
+    
+```rust
+struct Sender;
+
+impl Sender {
+    pub async fn try_payjoin(pj_uri: payjoin::Uri<'static, NetworkChecked>) -> Txid {
+    }
+}
+```
+
+
+In the next few steps, we will add the core implementation for the sender inside
+`try_payjoin` function.
+
+* Extract the parameters from the payjoin URI
+
+  Here we extract the amount the reciever is requesting and the address they
+  wish to recieve to. The URI also includes the endpoint we will make the
+  request to, and that is extracted in the `RequestContext::from_psbt_and_uri`
+  function and returned as part of the request data.
+
+```rust
+impl Sender {
+    pub async fn try_payjoin(pj_uri: payjoin::Uri<'static, NetworkChecked>) -> Txid {
+        let sender_wallet = get_client(Some("sender"));
+        let amount_to_send = pj_uri.amount.unwrap();
+        let receiver_address = pj_uri.address.clone();
+    }
+}
+```
+
+* Construct an "Original PSBT" paying `.amount` to `.address` as specified in
+  the URI.
+
+  First we add the reciever address to the outputs list along side the request
+  amount. Then we create a funded psbt with options set to specify `fee_rate`
+  and other paramters.
+
+```rust
+...
+pub async fn try_payjoin(pj_uri: payjoin::Uri<'static, NetworkChecked>) -> Txid {
+    ...
+    let mut outputs = HashMap::with_capacity(1);
+    outputs.insert(receiver_address.to_string(), amount_to_send);
+    let options = bitcoincore_rpc::json::WalletCreateFundedPsbtOptions {
+        lock_unspent: Some(false),
+        fee_rate: Some(bitcoincore_rpc::bitcoin::Amount::from_sat(10000)),
+        ..Default::default()
+    };
+    let sender_psbt = sender_wallet
+        .wallet_create_funded_psbt(
+            &[], // inputs
+            &outputs,
+            None, // locktime
+            Some(options),
+            None,
+        )
+        .unwrap();
+    let psbt = sender_wallet
+        .wallet_process_psbt(&sender_psbt.psbt, None, None, None)
+        .unwrap()
+        .psbt;
+    let psbt = Psbt::from_str(&psbt).unwrap();
+}
+```
+
+* Construct the HTTPS request
+
+  Prepare the request we want to send and also get the context back so we can
+  handle the response when its recieved.
```

Comment:
This is a good opportunity to explain why authenticated communication is necessary and plaintext is insufficient.